### PR TITLE
Set enterprise tier in backup documentation

### PR DESCRIPTION
**Pull request by hsadoyan**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind design

/kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:
Update example documentation to v1 from v1beta. 
Default to `tier: enterprise` instead of basic.
Update supported tier list in [backup.md](https://github.com/kubernetes-sigs/gcp-filestore-csi-driver/blob/master/docs/kubernetes/backup.md)

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
Update example documentation to v1 from v1beta
```

### CODE REVIEW

**Comment by hsadoyan** [comment]
This CL changes the backup documentation to default to `enterprise` tier.

If the other workflows are ready, I can change all of the example SCs to use `enterprise` as well

**Comment by amacaskill** [comment]
> This CL changes the backup documentation to default to `enterprise` tier.
> 
> If the other workflows are ready, I can change all of the example SCs to use `enterprise` as well

What other workflows are you referring to?
Also if this really is for documentation, can you please update the PR comment /kind to documentation?

**Comment by hsadoyan** [comment]
Ah I'd forgotten to add a newline before the documentation line. 

Under [examples/kubernetes](https://github.com/kubernetes-sigs/gcp-filestore-csi-driver/tree/master/examples/kubernetes/) we have configs for fsgroups, cross-region clusters, base demo SCs, and so on. Do we want to change all of them to enterprise?

**Comment by amacaskill** [comment]
> Ah I'd forgotten to add a newline before the documentation line.
> 
> Under [examples/kubernetes](https://github.com/kubernetes-sigs/gcp-filestore-csi-driver/tree/master/examples/kubernetes/) we have configs for fsgroups, cross-region clusters, base demo SCs, and so on. Do we want to change all of them to enterprise?

I'm not sure why we needed to set  enterprise tier in backup storageclass in the first place. Why are we making this change? I also don't think we want to change everything to enterprise because I think premier is what most people use.

**Comment by saikat-royc** [comment]
Can you also change the snapshot apis to v1

https://github.com/kubernetes-sigs/gcp-filestore-csi-driver/blob/master/examples/kubernetes/backups-restore/backup-volumesnapshotclass.yaml
https://github.com/kubernetes-sigs/gcp-filestore-csi-driver/blob/master/examples/kubernetes/backups-restore/backup.yaml
https://github.com/kubernetes-sigs/gcp-filestore-csi-driver/tree/master/examples/kubernetes/backups

and the documentation for the right v1 api and enterprise tier
https://github.com/kubernetes-sigs/gcp-filestore-csi-driver/blob/master/docs/kubernetes/backup.md

File: docs/kubernetes/backup.md [hunk 1]
```diff
@@ -29,6 +29,7 @@ The [CSI Snapshot](https://github.com/container-storage-interface/spec/blob/mast
     name: csi-filestore
     provisioner: filestore.csi.storage.gke.io
     parameters:
+      tier: enterprise
```
Comment (saikat-royc) [inline docs/kubernetes/backup.md:32]: Can you also fix any mention of supported tiers and add enterprise e.g https://screenshot.googleplex.com/Q6GvZxP3cosD5be

**Comment by saikat-royc** [comment]
Please put a short description in the PR "What this PR does / why we need it:"

File: docs/kubernetes/backup.md [hunk 2]
```diff
@@ -8,7 +8,7 @@ Kubernetes 1.17+. CSI VolumeSnapshot should not be confused with Filestore Backu
...
 
 ### Backup Example
-A Filestore backup is a copy of a file share that includes all file data and metadata of the file share from the point in time when the backup is created. It works for Basic HDD and Basic SSD tier instances. Once a backup of a file share is created, the original file share can be modified or deleted without affecting the backup. A file share can be completely restored from a backup as a new Filestore instance or onto an existing file share. For more details refer to this [documentation](https://cloud.google.com/filestore/docs/backups)
+A Filestore backup is a copy of a file share that includes all file data and metadata of the file share from the point in time when the backup is created. It works for Basic HDD, Basic SSD, and Enterprise tier instances. The tier and volume size of the backup must match the source volume. Once a backup of a file share is created, the original file share can be modified or deleted without affecting the backup. A file share can be completely restored from a backup as a new Filestore instance or onto an existing file share. For more details refer to this [documentation](https://cloud.google.com/filestore/docs/backups)
```
Comment (saikat-royc) [inline docs/kubernetes/backup.md:11]: link to the filestore tier page https://cloud.google.com/filestore/docs/service-tiers

Comment (hsadoyan) [reply docs/kubernetes/backup.md]: Done

**Pull request merged**